src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+        user.setPassword("password123");
+        user.setAge(20);
+        Storage.people.add(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin("validUser123");
+        duplicateUser.setPassword("password123");
+        duplicateUser.setAge(20);
+
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(duplicateUser));
+    }
+
+    @Test
+    void register_ageAboveMin_shouldAddUserToStorage() throws RegistrationException {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("validUser123");
+        user.setPassword("validPassword123");
+        user.setAge(20);
+
+        registrationService.register(user);
+
+        assertEquals(1, Storage.people.size());
+        assertEquals("validUser123", Storage.people.get(0).getLogin());
+    }
+
+    @Test
+    void register_passwordLengthAtMin_shouldSucceed() throws RegistrationException {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("userWithPass");
+        user.setPassword("pass12");
+        user.setAge(20);
+
+        registrationService.register(user);
+
+        assertEquals(1, Storage.people.size());
+        assertEquals("userWithPass", Storage.people.get(0).getLogin());
+    }
+
+    @Test
+    void register_passwordLengthBelowMin_shouldThrowException() {
+        User user = new User();
+        user.setLogin("userWithShortPass");
+        user.setPassword("pass");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullLogin_shouldThrowException() {
+        User user = new User();
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_shouldThrowException() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageBelowMin_shouldThrowException() {
+        User user = new User();
+        user.setLogin("underageUser");
+        user.setPassword("validPassword");
+        user.setAge(15);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_loginLengthAtMin_shouldSucceed() throws RegistrationException {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("abcde1");
+        user.setPassword("validPassword123");
+        user.setAge(20);
+
+        registrationService.register(user);
+
+        assertEquals(1, Storage.people.size());
+        assertEquals("abcde1", Storage.people.get(0).getLogin());
```

Comment:
This test case checks the successful registration of a user with a login length at the minimum. However, the checklist does not specify a minimum login length requirement. Make sure you are not adding rules that are not described in the task requirements.